### PR TITLE
feat(engine): integrate executor with StateRootTask

### PR DESCRIPTION
**Pull request by fgimenez**
Towards: #12053
Requires: #12316

Creates a `StateRootTask` instance and wires it with the state hook passed to the executor using an unbounded channel. For now the `StateRootTask` instance is not created yet, some TODOs added for when all the functionality is in place and we can switch to use it.

### CODE REVIEW

File: crates/engine/tree/src/tree/mod.rs [hunk 1]
```diff
@@ -2188,16 +2189,31 @@ where
...
+        let state_root_task = StateRootTask::new(
+            consistent_view,
+            // TODO: calculate `TrieInput` like in `self.compute_state_root_parallel`.
+            Arc::new(TrieInput::default()),
+            UnboundedReceiverStream::new(state_root_rx),
+        );
+        let state_root_handle = state_root_task.spawn();
+        let state_hook = move |result_and_state: &ResultAndState| {
+            let _ = state_root_tx.send(result_and_state.state.clone());
+        };
```
Comment (rkrasiuk) [inline crates/engine/tree/src/tree/mod.rs:2206]: let's leave the integration for once we finish with the task? i'd prefer to have it ready before enabling it in the tree

Comment (fgimenez) [reply crates/engine/tree/src/tree/mod.rs]: makes total sense, done here https://github.com/paradigmxyz/reth/pull/12335/commits/ba4a80401b0108514939be04161f7ad4e113b459 i'll add some unit tests on a follow up to cover the existing functionality

File: crates/engine/tree/src/tree/root.rs [hunk 2]
```diff
@@ -4,23 +4,37 @@ use reth_provider::providers::ConsistentDbView;
...
+#[allow(dead_code)]
+pub(crate) struct StateRootHandle {
+    /// Channel for receiving the final result.
+    rx: mpsc::Receiver<StateRootResult>,
+}
+
+#[allow(dead_code)]
+impl StateRootHandle {
+    /// Waits for the state root calculation to complete.
+    pub(crate) fn wait_result(self) -> StateRootResult {
```
Comment (rkrasiuk) [inline crates/engine/tree/src/tree/root.rs:25]: nit
```suggestion
    pub(crate) fn wait_for_result(self) -> StateRootResult {
```

File: crates/engine/tree/src/tree/root.rs [hunk 3]
```diff
@@ -41,20 +58,45 @@ impl<Factory> StateRootTask<Factory> {
...
 
-    fn poll(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<Self::Output> {
         // TODO:
-        //    * poll incoming state updates stream
         //    * keep track of proof calculation
         //    * keep track of intermediate root computation
-        Poll::Pending
+        //    * return final state root result
+        Ok((B256::default(), TrieUpdates::default()))
     }
```
Comment (rkrasiuk) [inline crates/engine/tree/src/tree/root.rs:101]: we'll do a bunch of polling & spawning here, i'd prefer to keep this a future

Comment (fgimenez) [reply crates/engine/tree/src/tree/root.rs]: sre done here https://github.com/paradigmxyz/reth/pull/12335/commits/53582dd340edbe4118a76ad74045e5bea890f117 PTAL

**Review by rkrasiuk — APPROVED** [review]

**Pull request merged**